client/js/app/components/explorer/visualization/keen_viz.js

```diff
@@ -35,6 +35,9 @@ var KeenViz = React.createClass({
   // ***********************
 
   shouldComponentUpdate: function(nextProps, nextState) {
+    if (this.props.model.metadata.visualization.chart_type === "table") {
```

Comment:
Updating the entire table if not required isn't the best scenario either. You may want to check if the extraction fields have changed instead to get an accurate idea of whether we really need to update this component. Re-rendering an entire table adds to the sluggishness of the app overall.